API.adoc

```diff
@@ -0,0 +1,611 @@
+
+=== Role Authorities
+
+- https://authorities.smartcosmos.net/roles/create
+- https://authorities.smartcosmos.net/roles/read
+- https://authorities.smartcosmos.net/roles/update
+- https://authorities.smartcosmos.net/roles/delete
+
+== REST API
+
+<<tenants, *Available methods for tenants*>>
+
+|====================
+|Endpoint|REST API|Required Authority
+|<<tenantcreate1, Create>>|`POST /tenants`|none
+|<<tenantupdate1, Update>>|`PUT /tenants/{urn}`|https://authorities.smartcosmos.net/tenants/update
+|<<tenantread1, Find by URN>>|`GET /tenants/{urn}`|https://authorities.smartcosmos.net/tenants/read
+|<<tenantread2, Find by Name>>|`GET /tenants/?name={name}`|https://authorities.smartcosmos.net/tenants/read
+|====================
+
+<<users, *Available methods for users*>>
+
+|====================
+|Endpoint|REST API|Required Authority
+|<<usercreate1, Create>>|`POST /users`|https://authorities.smartcosmos.net/users/create
+|<<userupdate1, Update>>|`PUT /users/{urn}`|https://authorities.smartcosmos.net/users/update
+|<<userread1, Find by URN>>|`GET /users/{urn}`|https://authorities.smartcosmos.net/users/read
+|<<userread2, Find by Name>>|`GET /users/?name={name}`|https://authorities.smartcosmos.net/users/read
+|<<userdelete1, Delete>>|`DELETE /users/{urn}`|https://authorities.smartcosmos.net/users/delete
+|====================
+
+<<roles, *Available methods for roles*>>
+
+|====================
+|Endpoint|REST API|Required Authority
+|<<rolecreate1, Create>>|`POST /roles`|https://authorities.smartcosmos.net/roles/create
+|<<roleupdate1, Update>>|`PUT /roles/{urn}`|https://authorities.smartcosmos.net/roles/update
+|<<roleread1, Find by URN>>|`GET /roles/{urn}`|https://authorities.smartcosmos.net/roles/read
+|<<roleread2, Find by Name>>|`GET /roles/?name={name}`|https://authorities.smartcosmos.net/roles/read
+|<<roledelete1, Delete>>|`DELETE /roles/{urn}`|https://authorities.smartcosmos.net/roles/delete
+|====================
+
+=== JSON Fields for Tenants
+[width="100%",options="header",cols="1,^1,2,^1,5"]
+|====================
+| Field | Format | Default | Required | Description
+| `urn` | String representation of a tenant identifier in a common scheme, e.g. `urn:tenant:uuid:<UUID>` | generated, value and format depend on the persistence layer implementation | | will be generated
+| `name` | String | | yes | Name of the tenant organization (e.g. `Example Company`)
+| `active` | boolean |  true | no |
+| `username` | String | | yes (for create) | Initial admin account
+|====================
+
+=== JSON Fields for Users
+[width="100%",options="header",cols="1,^1,2,^1,5"]
+|====================
+| Field | Format | Default | Required | Description
+| `urn` | String representation of a user identifier in a common scheme, e.g. `urn:user:uuid:<UUID>` | generated, value and format depend on the persistence layer implementation | | will be generated
+| `username` | String | | yes | login name, e.g. `bob@example.com`
```

Comment:
IMHO we should dump the email address and require the username to be parseable as an email address. But as Will points out, it's too late now.